src/Symfony/Component/Translation/Bridge/Loco/LocoProvider.php

```diff
@@ -91,92 +84,137 @@ public function read(array $domains, array $locales): TranslatorBag
             foreach ($domains as $domain) {
-                $response = $this->client->request('GET', sprintf('export/locale/%s.xlf?filter=%s&status=translated', $locale, $domain));
+                $responses[] = [
+                    'response' => $this->client->request('GET', sprintf('export/locale/%s.xlf?filter=%s&status=translated', $locale, $domain)),
+                    'locale' => $locale,
+                    'domain' => $domain,
+                ];
+            }
+        }
 
-                if (404 === $response->getStatusCode()) {
-                    $this->logger->error(sprintf('Locale "%s" for domain "%s" does not exist in Loco.', $locale, $domain));
-                    continue;
-                }
+        foreach ($responses as $response) {
+            $locale = $response['locale'];
+            $domain = $response['domain'];
+            $response = $response['response'];
 
-                $responseContent = $response->getContent(false);
+            if (404 === $response->getStatusCode()) {
+                $this->logger->error(sprintf('Locale "%s" for domain "%s" does not exist in Loco.', $locale, $domain));
+                continue;
+            }
 
-                if (200 !== $response->getStatusCode()) {
-                    throw new ProviderException('Unable to read the Loco response: '.$responseContent, $response);
-                }
+            $responseContent = $response->getContent(false);
 
-                $translatorBag->addCatalogue($this->loader->load($responseContent, $locale, $domain));
+            if (200 !== $response->getStatusCode()) {
+                throw new ProviderException('Unable to read the Loco response: '.$responseContent, $response);
             }
+
+            $translatorBag->addCatalogue($this->loader->load($responseContent, $locale, $domain));
         }
 
         return $translatorBag;
     }
 
     public function delete(TranslatorBagInterface $translatorBag): void
     {
-        $deletedIds = [];
+        $catalogue = $translatorBag->getCatalogue($this->defaultLocale);
 
-        foreach ($translatorBag->getCatalogues() as $catalogue) {
-            foreach ($catalogue->all() as $messages) {
-                foreach ($messages as $id => $message) {
-                    if (\in_array($id, $deletedIds, true)) {
-                        continue;
-                    }
-
-                    $this->deleteAsset($id);
-                    $deletedIds[] = $id;
-                }
+        if (!$catalogue) {
+            $catalogue = $translatorBag->getCatalogues()[0];
+        }
+
+        $responses = [];
+
+        foreach (array_keys($catalogue->all()) as $domain) {
+            foreach ($this->getAssetsIds($domain) as $id) {
+                $responses[$id] = $this->client->request('DELETE', sprintf('assets/%s.json', $id));
+            }
+        }
+
+        foreach ($responses as $key => $response) {
+            if (200 !== $response->getStatusCode()) {
```

Comment:
what about 404 responses ? Should these be accepted as already deleted ?

Reply:
Already deleted or not found, and so reputed already deleted. In all case we should ignore them.